Feature/13041

djordjedjukic opened the pull request.

arekpalinski commented on src/Raven.Client/Documents/Operations/ConnectionStrings/GetConnectionStringsOperation.cs:
```diff
@@ -80,11 +81,13 @@ public class GetConnectionStringsResult
         public Dictionary<string, RavenConnectionString> RavenConnectionStrings { get; set; }
         public Dictionary<string, SqlConnectionString> SqlConnectionStrings { get; set; }
         public Dictionary<string, OlapConnectionString> OlapConnectionStrings { get; set; }
+        public Dictionary<string, ElasticsearchConnectionString> ElasticConnectionStrings { get; set; }
```
ElasticConnectionStrings -> ElasticsearchConnectionStrings

Just to follow the naming convention

arekpalinski commented on src/Raven.Client/Documents/Operations/ETL/Elasticsearch/ElasticsearchEtlConfiguration.cs:
```diff
@@ -0,0 +1,61 @@
+
+        public override string GetDefaultTaskName()
+        {
+            return $"Elastic ETL to {ConnectionStringName}";
```
Elastic -> Elasticsearch

arekpalinski commented on src/Raven.Server/Documents/ETL/Providers/Elasticsearch/ElasticsearchDocumentTransformer.cs:
```diff
@@ -0,0 +1,122 @@
+
+            GetOrAdd(indexName).Inserts.Add(new ElasticsearchItem(Current) {Property = property});
+
+            //_stats.IncrementBatchSize(result.Size);
```
Commented?

arekpalinski commented on src/Raven.Server/Documents/ETL/Providers/Elasticsearch/ElasticsearchEtl.cs:
```diff
@@ -0,0 +1,160 @@
+                        index: index.IndexName.ToLower(),
+                        body: insert.Property.RawValue.ToString(), requestParameters: new IndexRequestParameters(){Refresh = Refresh.WaitFor});
+
+                    if (!response.Success)
```
Minor: please convert to `if (response.Success == false)`. We always use this notation

arekpalinski commented on src/Raven.Server/Documents/ETL/Providers/Elasticsearch/ElasticsearchEtl.cs:
```diff
@@ -0,0 +1,160 @@
+
+                if (deleteResponse.ServerError != null)
+                {
+                    throw new Exception(deleteResponse.ServerError.Error.ToString());
```
Can we have a test were we put _incorrect_ URLs of Elasticsearch and see the this generates an alert about load error?

See: https://github.com/ravendb/ravendb/blob/e36526d2c4c8a13fecf6dcea2c4dd47a53c3c8e7/src/Raven.Server/NotificationCenter/EtlNotifications.cs#L34-L43

arekpalinski commented:
@aviv86 Could you please do the review?

aviv86 commented on src/Raven.Server/Documents/ETL/Providers/Elasticsearch/ElasticsearchDocumentTransformer.cs:
```diff
@@ -0,0 +1,122 @@
+
+        protected override void AddLoadedAttachment(JsValue reference, string name, Attachment attachment)
+        {
+            throw new System.NotImplementedException();
```
shouldn't this be `NotSupportedException` ?

aviv86 commented on src/Raven.Server/ServerWide/ClusterStateMachine.cs:
```diff
@@ -404,16 +404,19 @@ protected override void Apply(ClusterOperationContext context, BlittableJsonRead
                     case nameof(UpdateRavenEtlCommand):
                     case nameof(UpdateSqlEtlCommand):
                     case nameof(UpdateOlapEtlCommand):
                     case nameof(DeleteOngoingTaskCommand):
```
there's no `Update` command for Elasticsearch ETL?

aviv86 commented on src/Raven.Client/Documents/Operations/ETL/Elasticsearch/ElasticsearchEtlConfiguration.cs:
```diff
@@ -0,0 +1,61 @@
+
+        protected bool Equals(ElasticsearchIndex other)
+        {
+            return string.Equals(IndexName, other.IndexName) && string.Equals(IndexIdProperty, other.IndexIdProperty, StringComparison.OrdinalIgnoreCase);
```
`IndexName` is case sensitive ?

aviv86 commented on test/SlowTests/Server/Documents/ETL/Elasticsearch/ElasticsearchEtlTests.cs:
```diff
@@ -0,0 +1,822 @@
+                        }
+                    }
+                },
+                new ElasticsearchConnectionString {Name = connectionStringName, Nodes = new[] {"http://localhost:9200"}});
```
how are these tests going to work in our CI ?
don't we need to spin up elastic instance in `localhost:9200` ?
or am I missing something ?

arekpalinski replied:
We'll take care of setting up our CI accordingly on our own.

djordjedjukic committed:
RavenDB-13041 Elasticsearch ETL

RavenDB-13041 fixing compilation

rename Elastic to Elasticsearch


implement CreateScope method


modify RemoveConnectionString method


remove unused code


add missing part in license status


PR comments resolved

arekpalinski commented:
Closing this PR as we opened it against main RavenDB repo - https://github.com/ravendb/ravendb/pull/12452
